src/vegarando.js

```diff
@@ -1,8 +1,35 @@
 "use strict";
 
-const bad = /aal|salami|bacon|sucuk|schinken|chicken|ente|bulette|fisch|garnele|meeresfr|leber|kalb|schwein|lachs|pastete|lamm|prosciutto|kapern|sardellen|gyros|Hähnchen|Hänchen|hühner|hühn|hünchen|huhn|Scampi|wurst|rind|beef|fleisch|krabben|hackfleisch|wurst|würstchen|schnitzel|steak|turkey|kebab|kebap|pute|speck|shrimps|scmpi|bolognese|ribs|rippchen|rogan josh|goshet|murgh|döner|frutti di mare|prosciutto|gehacktes|gehacktem|wiener art|burger|ebi|kani|surimi|sake|unagi|tako|masago|saba|tekka|maguro|mongoika/gim;
-const good = /vegetarisch|vegan|veggie|vegetaria|chay/gim; // chay is vietnamese for vegetarian
-const vegan = /vegan/gim;
+/**
+ * Search a given haystack string for an occurence of any of the keywords
+ * given in `needles` and return any matches
+ * @param {string} haystack
+ * @param {string[]} needles
+ * @return {string[]} List of matching keywords in original casing as present in haystack
+ */
+function getMatchingKeywords(haystack, needles) {
+    let matches = new Set();
+
+    const regex = new RegExp(needles.join("|"), "gim");
+
+    for (let match of haystack.matchAll(regex)) {
+        matches.add(match[0]);
+    }
+
+    return Array.from(matches);
+}
+
+/**
+ * Check if the haystack string contains any of the needles
+ * @param {string} haystack
+ * @param {string[]} needles
+ * @return {boolean}
+ */
+function hasMatchingKeyword(haystack, needles) {
+    const regex = new RegExp(needles.join("|"), "gim");
```

Comment:
this would compile a new regex on every function call, probably best to compile these only regexps for every list

Reply:
You're absolutely correct. I initially wanted to go with an array matching approach but was afraid of breaking default behaviour. I'll refactor it slightly to generate the expression once.